xla/stream_executor/sycl/sycl_gpu_runtime.cc

```diff
@@ -38,6 +37,96 @@ absl::Status IsValidDeviceOrdinal(int device_ordinal,
   }
 }
 
+// Returns true if the oneAPI version is 2024.2 or newer.
+// oneAPI 2024.2 corresponds to __LIBSYCL_MAJOR_VERSION == 7 and
+// __LIBSYCL_MINOR_VERSION == 2.
+bool IsOneAPIVersionAtLeast2024_2() {
+  return (__LIBSYCL_MAJOR_VERSION >= 7) && (__LIBSYCL_MINOR_VERSION >= 2);
+}
+
+absl::Status MemcpyDeviceToHost(::sycl::queue* stream_handle, void* dst_host,
+                                const void* src_device, size_t byte_count,
+                                bool async = false) {
+  try {
+    ::sycl::event event =
+        stream_handle->memcpy(dst_host, src_device, byte_count);
+    if (!async) {
+      event.wait();
+    }
+  } catch (const ::sycl::exception& e) {
+    return absl::InternalError(
+        "MemcpyDeviceToHost failed: " + std::string(e.what()) +
+        ", file = " + __FILE__ + ", line = " + std::to_string(__LINE__) + ".");
+  }
+  return absl::OkStatus();
+}
+
+absl::Status MemcpyHostToDevice(::sycl::queue* stream_handle, void* dst_device,
+                                const void* src_host, size_t byte_count,
+                                bool async = false) {
+  try {
+    ::sycl::event event =
+        stream_handle->memcpy(dst_device, src_host, byte_count);
+    if (!async) {
+      event.wait();
+    }
+  } catch (const ::sycl::exception& e) {
+    return absl::InternalError(
+        "MemcpyHostToDevice failed: " + std::string(e.what()) +
+        ", file = " + __FILE__ + ", line = " + std::to_string(__LINE__) + ".");
+  }
+  return absl::OkStatus();
+}
+
+absl::Status MemcpyDeviceToDevice(::sycl::queue* stream_handle,
+                                  void* dst_device, const void* src_device,
+                                  size_t byte_count, bool async = false) {
+  try {
+    ::sycl::event event =
+        stream_handle->memcpy(dst_device, src_device, byte_count);
+    if (!async) {
+      event.wait();
+    }
+  } catch (const ::sycl::exception& e) {
+    return absl::InternalError(
+        "MemcpyDeviceToDevice failed: " + std::string(e.what()) +
+        ", file = " + __FILE__ + ", line = " + std::to_string(__LINE__) + ".");
+  }
+  return absl::OkStatus();
+}
+
+absl::Status MemsetDevice(::sycl::queue* stream_handle, void* dst_device,
+                          unsigned char value, size_t count,
+                          bool async = false) {
+  try {
+    ::sycl::event event =
+        stream_handle->memset(dst_device, value, count * sizeof(uint8_t));
+    if (!async) {
+      event.wait();
+    }
+  } catch (const ::sycl::exception& e) {
+    return absl::InternalError("MemsetDevice failed: " + std::string(e.what()) +
+                               ", file = " + __FILE__ +
+                               ", line = " + std::to_string(__LINE__) + ".");
+  }
+  return absl::OkStatus();
+}
+
+absl::Status MemfillDevice(::sycl::queue* stream_handle, void* dst_device,
```

Comment:
Can you either make `stream_handle` a reference instead of a pointer (preferred) or add the `absl_nonnull` attribute?

Reply:
Done.